content/docs/command-reference/run.md

```diff
@@ -20,28 +22,56 @@ positional arguments:
+`dvc run` provides an interface to describe
+[pipelines](/doc/command-reference/pipeline), stage by stage. _Stages_ represent
+individual data processes, including their input and resulting outputs.
+
+A stage name is required and should be provided using the `-n` (`--name`)
+option. Stages are stored in a YAML pipelines file named
+[`dvc.yaml`](/doc/user-guide/dvc-files-and-directories) using this name. The
+other available [options](#options) are mainly meant to describe the stage
+dependencies (input) an output files or directories.
 
 The remaining terminal input provided to `dvc run` after the command options
 (`-`/`--` flags) will become the required `command` argument. Please wrap the
 `command` with `"` quotes if there are special characters in it like `|` (pipe)
-or `<`, `>` (redirection) that would otherwise apply to `dvc run` itself e.g.
-`dvc run -d script.sh "./script.sh > /dev/null 2>&1"`. Use single quotes `'`
-instead of `"` to wrap the `command` if there are environment variables in it,
-that you want to be evaluated dynamically. E.g.
-`dvc run -d script.sh './myscript.sh $MYENVVAR'`
-
-Since `dvc run` provides a way to build a dependency graph using dependencies
-and outputs to connect different stages, it checks the graph's integrity before
-creating a new stage. For example, for every output there should be only one
-stage that explicitly specifies it, there should be no cycles, etc.
+or `<`, `>` (redirection) e.g.
+`dvc run -n mystage "./script.sh > /dev/null 2>&1"` (otherwise they would apply
+to `dvc run` itself). Use single quotes `'` instead of `"` to wrap the `command`
+if there are environment variables in it that should be evaluated dynamically
+e.g. `dvc run -n mystage './myscript.sh $MYENVVAR'`
+
+### Dependencies and outputs
+
+By specifying lists of <abbr>dependencies</abbr> (`-d` option) and/or
+<abbr>outputs</abbr> (`-o` and `-O` options) for each stage, we can outline a
+_dependency graph_ ([DAG](https://en.wikipedia.org/wiki/Directed_acyclic_graph))
+that connects them, i.e. the output of a stage becomes the input of another, and
+so on. This graph can be restored by DVC later to modify or
+[reproduce](/doc/command-reference/repro) the full
+[pipeline](/doc/command-reference/pipeline).
+
+`dvc run` checks the dependency graph integrity before creating a new stage. For
+example, for every output there should be only one stage that explicitly
+specifies it, there should be no cycles, etc.
+
+A special type of key/value dependencies,
+[parameters](/doc/command-reference/params) (`-p` option), is supported.
+Multiple parameter dependencies can be specified from one or more YAML or JSON
+parameters files (`params.yaml` by default).
+
+Special types of output files, [metrics](/doc/command-reference/metrics) (`-m`
+and `-M` options) and [plots](/doc/command-reference/plots), are also supported.
+Metrics files have specific formats (JSON, YAML, CSV, or TSV) and allow for
```

Comment:
Note I'm using "metrics files" here @shcheklein. I have been thinking about it and I think we should consider always using plural "metrics" for everything. We don't use "Metric file" because it sounds strange. Well, I think "metric files" has the same problem: "metric" out of context sounds like we're talking about a measure of distance (as in kilometers).

And BTW I also think we shouldn't call plots files metrics files which we do now because technically they are metrics, but always call them plots files.

WDYT? I would extract this to a separate issue of course. It may even impact the core repo.

Reply:
I would say, it's definitely not the right time to discuss this. Let's do it consistent in some way. Otherwise up to you regarding the ticker or changing it everywhere - I'm fine to have it either way (though `metrics files` sound like wrong from the language perspective). You might ask @andronovhopf as well, btw.

Reply:
plots vs metrics - I'm fine with either for now

Reply:
> metrics files sound like wrong from the language perspective

A little bit but we also talk about "pipelines file" which would become "pipelines files" if pluralized, I think it's fine.

Anyway yes, totally not the right place to discuss this, you're right. I'll change this instance to singular for consistency and open a separate ticket, ping Dmitry and Elle.

Reply:
Extracted to #1442

Reply:
But I don't think I can roll this back to "metric files" because it's really out of context. Even when the concept of "metrics" is introduced in the same paragraph, it's not obvious what "metric files" are, whereas "metrics files" makes more sense since it's the exact same word and there's no ambiguity, so you can relate the concepts.

Reply:
Actually I'm going to go with "Metrics and plots files" here anyway, ha. See 5962ce06